scripts/copyrighter.py

```diff
@@ -0,0 +1,156 @@
+        return [f"*{e}" for e in self.extensions]
+
+    def _write_copyright_header(
+        self, file_contents: list[str], has_shebang: bool
+    ) -> list[str]:
+        new_content: list[str] = []
+        if has_shebang:
+            new_content += file_contents[0:2]
+            new_content += self.comment
+            if len(file_contents) >= 2:
+                new_content += file_contents[2:]
+        else:
+            new_content = self.comment.copy()
+            new_content += file_contents
+
+        return new_content
+
+    def _update_copyright_header(
+        self, file_contents: list[str], has_shebang: bool
+    ) -> list[str]:
+        new_content: list[str] = []
+        if has_shebang:
+            new_content += file_contents[0:2]
+            new_content.append(self.comment[0])
+            if len(file_contents) >= 4:
+                new_content += file_contents[3:]
+        else:
+            new_content.append(self.comment[0])
+            new_content += file_contents[1:]
+        return new_content
+
+    def _has_copyright_header(
+        self, file_contents: list[str], has_shebang: bool
+    ) -> bool:
+        if has_shebang:
+            return (
+                len(file_contents) >= 4
+                and self.header_pattern.search(file_contents[2]) != None
+            )
+        return (
+            len(file_contents) >= 2
+            and self.header_pattern.search(file_contents[0]) != None
+        )
+
+    def handle(self, file: Path):
+        if not any([file.name.endswith(e) for e in self.extensions]):
+            return
+
+        file_contents = file.read_text().splitlines()
+        has_shebang = (
+            len(file_contents) >= 1
+            and self.account_for_shebang
+            and file_contents[0].startswith("#!")
+        )
+        new_content = None
+        if self._has_copyright_header(file_contents, has_shebang):
+            new_content = self._update_copyright_header(
+                file_contents, has_shebang
+            )
+        else:
+            new_content = self._write_copyright_header(
+                file_contents, has_shebang
+            )
+        if new_content[-1] != "\n":
+            new_content.append("\n")
+        file.write_text("\n".join(new_content))
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-e",
+        "--exclude",
+        type=str,
+        action="append",
+        help="directories (path prefixes) to exclude",
+    )
+    parser.add_argument(
+        "-y",
+        "--year",
+        type=int,
+        help="New end year for copyright header",
+        default=datetime.date.today().year,
+    )
+    parser.add_argument("root", type=Path, default=Path.cwd())
+    return parser.parse_args()
+
+
+def main():
+    args = parse_arguments()
+    type_handlers = [
+        Handler(args.year, [".py"], "# {}", True),
+        Handler(args.year, [".hpp", ".cpp"], "// {}", False),
```

Comment:
Before cpp copyright header started with `///`. Is there a reason for the change?